### PR TITLE
optimize: refactor lock store sql with SPI

### PR DESCRIPTION
**Pull request by objcoding**
### Ⅰ. Describe what this PR did
Refactor lock store sql with spi.

### CODE REVIEW

**Comment by l81893521** [comment]
Please add unit test.

**Comment by objcoding** [comment]
> Please add unit test.

done.

**Comment by l81893521** [comment]
Please resolve the code conflict.

**Comment by objcoding** [comment]
> Please resolve the code conflict.

done.

File: core/src/main/java/io/seata/core/store/db/sql/lock/MysqlLockStoreSql.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @since 1.2.0
+ */
+@LoadLevel(name = "mysql")
+public class MysqlLockStoreSql extends AbstractLockStoreSql {
+
+    /**
+     * The constant INSERT_LOCK_SQL_MYSQL.
+     */
+    private static final String INSERT_LOCK_SQL_MYSQL = "insert into " + LOCK_TABLE_PLACE_HOLD + "(" + ALL_COLUMNS + ")" +
+        "values (?, ?, ?, ?, ?, ?, ?, now(), now())";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/store/db/sql/lock/MysqlLockStoreSql.java]: notice the space

File: core/src/main/java/io/seata/core/store/db/sql/lock/OceanbaseLockStoreSql.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @since 1.2.0
+ */
+@LoadLevel(name = "oceanbase")
+public class OceanbaseLockStoreSql extends AbstractLockStoreSql {
+
+    /**
+     * The constant INSERT_LOCK_SQL_OCEANBASE.
+     */
+    private static final String INSERT_LOCK_SQL_OCEANBASE = "insert into " + LOCK_TABLE_PLACE_HOLD + "(" + ALL_COLUMNS + ")" +
+        "values (?, ?, ?, ?, ?, ?, ?, now(), now())";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/store/db/sql/lock/OceanbaseLockStoreSql.java]: notice the space

File: core/src/main/java/io/seata/core/store/db/sql/lock/OracleLockStoreSql.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+@LoadLevel(name = "oracle")
+public class OracleLockStoreSql extends AbstractLockStoreSql {
+
+    /**
+     * The constant INSERT_LOCK_SQL_ORACLE.
+     */
+    private static final String INSERT_LOCK_SQL_ORACLE = "insert into " + LOCK_TABLE_PLACE_HOLD + "(" + ALL_COLUMNS + ")"
+        +
+        "values (?, ?, ?, ?, ?, ?, ?, sysdate, sysdate)";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/store/db/sql/lock/OracleLockStoreSql.java]: notice the space

File: core/src/main/java/io/seata/core/store/db/sql/lock/PostgresqlLockStoreSql.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+@LoadLevel(name = "postgresql")
+public class PostgresqlLockStoreSql extends AbstractLockStoreSql {
+
+    /**
+     * The constant INSERT_LOCK_SQL_POSTGRESQL.
+     */
+    private static final String INSERT_LOCK_SQL_POSTGRESQL = "insert into " + LOCK_TABLE_PLACE_HOLD + "(" + ALL_COLUMNS + ")"
+        +
+        "values (?, ?, ?, ?, ?, ?, ?, now(), now())";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/store/db/sql/lock/PostgresqlLockStoreSql.java]: notice the space

File: core/src/main/java/io/seata/core/store/db/sql/lock/MysqlLockStoreSql.java (hunk 1 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/store/db/sql/lock/MysqlLockStoreSql.java]: done.

File: core/src/main/java/io/seata/core/store/db/sql/lock/OceanbaseLockStoreSql.java (hunk 2 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/store/db/sql/lock/OceanbaseLockStoreSql.java]: done.

File: core/src/main/java/io/seata/core/store/db/sql/lock/PostgresqlLockStoreSql.java (hunk 4 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/store/db/sql/lock/PostgresqlLockStoreSql.java]: done.

**Pull request merged**